Trie/prefix_tree.rb

```diff
@@ -0,0 +1,111 @@
+require_relative "prefix_tree_node"
+
+class PrefixTree
+  attr_accessor :root, :container
+
+  def initialize
+    @root = Node.new
+    @container = []
+  end
+
+  def add(word)
+    container << word
+    current = root
+    word.each_char do |char|
+      current = adder_logic(current, char)
+    end
+    current.is_word = true
+    current
+  end
+
+  def adder_logic(current, char)
+    if current.data.include?(char)
+      current = current.data[char]
+    else
+      new_node = Node.new
+      current.data[char] = new_node
+      current = new_node
+    end
+    current
+  end
+
+  def include?(prefix)
+    current = root
+    prefix.each_char do |char|
+      return false unless current.data.include?(char)
+      current = current.data[char]
+    end
+    true
+  end
+
+  def find(prefix)
+    current = root
+    prefix.each_char do |char|
+      return false unless current.data.include?(char)
+      current = current.data[char]
+    end
+    current.is_word
+  end
+
+  def list(prefix = nil)
+    return container if prefix.nil?
+    verified_words(container, prefix)
+  end
+
+  def verified_words(container, prefix)
+    res = []
+    container.each do |word|
+      res << word if word.start_with?(prefix)
+    end
+    res.length == 0 ? false : res
+  end
+
+  def delete(word)
```

Comment:
method delete is not wor
<img width="939" alt="Screenshot 2022-05-04 at 12 50 11" src="https://user-images.githubusercontent.com/6873183/166667801-c0cdc530-6fab-4297-8730-d8967fe30ae2.png">
k

Reply:
fixed